README.md

```diff
@@ -1,2 +1,151 @@
+
+# Activate the machine
+eval "$(docker-machine env jupyterhub)"
+```
+
+Docker Machine can also provision new hosts using one of it's [supported drivers](https://docs.docker.com/machine/drivers/).   When provisioning a host, Docker Machine will automatically generate TLS certificate and key files and use them to authenticate with the remote Docker daemon.
+
+## Create a Docker Network
+
+Create a Docker network for inter-container communication.  The benefits of using a Docker network are:
+
+* container isolation - only the containers on the network can access one another
+* name resolution - Docker daemon runs an embedded DNS server to provide automatic service discovery for containers connected to user-defined networks.  This allows us to access containers on the same network by name.
+
+Here we create a Docker network named `jupyterhub-network`.  Later, we will configure the JupyterHub and single-user Jupyter Notebook containers to run attached to this network.
+
+```
+docker network create jupyterhub-network
+```
+
+## Setup GitHub Authentication
+
+This deployment uses GitHub OAuth to authenticate users.  It requires that you create a [GitHub application](https://github.com/settings/applications/new). You will need to specify an OAuth callback URL in the following form:
+
+```
+https://<myhost.mydomain>/hub/oauth_callback
+```
+
+You must pass the secrets that GitHub provides for your application to JupyterHub at runtime.  You can do this by setting the `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET`, and `OAUTH_CALLBACK_URL` environment variables when you run the JupyterHub container, or you can add them to the `.env` file in the root directory of this repository.  For example,
+
+```
+GITHUB_CLIENT_ID=<github_client_id>
+GITHUB_CLIENT_SECRET=<github_client_secret>
+OAUTH_CALLBACK_URL=https://<myhost.mydomain>/hub/oauth_callback
+```
+
+## Build the JupyterHub Docker image
+
+Configure JupyterHub and build it into a Docker image.
+
+1. Copy your TLS certificate and key to a directory named `secrets` within this repository directory.  These will be added to the Docker image at build time.
+
+    ```
+    mkdir -p secrets
+    cp jupyterhub.cer jupyterhub.key secrets/
+    ```
+
+1. Create a `userlist` file with a list of authorized users.  At a minimum, this file should contain a single admin user.  The username should be a GitHub username.  For example:
+
+  	```
+  	jtyberg admin
+  	```
+
+	  The admin user will have the ability to add more users in the JupyterHub admin console.
+
+1. Build the JupyterHub Docker image.  For convenience, this repo provides a `hub.sh` script that wraps [docker-compose](https://docs.docker.com/compose/reference/), so you can run it with the docker-compose [command line arguments](https://docs.docker.com/compose/reference/overview/).  To build the JupyterHub image on the active Docker machine host, run:
+
+    ```
+    ./hub.sh build
+    ```
+
+## Create a JupyterHub Data Volume
```

Comment:
It might be good to include something here (perhaps external) about backing up this data volume.